services/common/src/test/java/org/collectionspace/services/common/test/SecurityUtilsTest.java

```diff
@@ -0,0 +1,222 @@
+    	
+		return testAssertion;
+    }
+    private Assertion createTestAssertionUntypedAttributeValues() throws IllegalArgumentException, IllegalAccessException, NoSuchFieldException, SecurityException {
+		Assertion testAssertion = createTestAssertionNoAttributes();
+
+		Attribute attr = createAttribute(false);
+		
+		AttributeStatement attrStmt = createNewSAMLObject(AttributeStatement.class);
+		attrStmt.getAttributes().add(attr);
+		testAssertion.getAttributeStatements().add(attrStmt);
+    	
+		return testAssertion;
+    }
+
+    // the tests are below
+    private Assertion testAssertionTypedAttributeValues = null;
+    private Assertion testAssertionUntypedAttributeValues = null;
+	@BeforeSuite
+    private void setup() throws InitializationException,NoSuchFieldException,IllegalAccessException {
+    	// try to set up openSAML
+		XMLObjectProviderRegistry registry = new XMLObjectProviderRegistry();
+		ConfigurationService.register(XMLObjectProviderRegistry.class, registry);
+		try {
+			InitializationService.initialize();
+		} catch (InitializationException e) {
+			logger.error("Could not initialize openSAML: " + e.getLocalizedMessage(), e);
+			throw e;
+		}	
+		// try to create a test assertion with typed attribute values; fail the test if this doesn't work
+		try {
+			testAssertionTypedAttributeValues = createTestAssertionTypedAttributeValues();
+		} catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException | SecurityException e) {
+			logger.error("Could not create test assertion with typed attribute values: " + e.getLocalizedMessage(), e);
+			throw e;
+		}
+		// try to create a test assertion with untyped attribute values; fail the test if this doesn't work
+		try {
+			testAssertionUntypedAttributeValues = createTestAssertionUntypedAttributeValues();
+		} catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException | SecurityException e) {
+			logger.error("Could not create test assertion with untyped attribute values: " + e.getLocalizedMessage(), e);
+			throw e;
+		}
+    }
+    
+    @Test
+    public void assertionWithTypedAttributeValuesIsNotNull() {
+    	testBanner("the mock assertion with typed attribute values is not null");
+    	Assert.assertNotNull(testAssertionTypedAttributeValues);
+    }
+    @Test
+    public void assertionWithUntypedAttributeValuesIsNotNull() {
+    	testBanner("the mock assertion with untyped attribute values is not null");
+    	Assert.assertNotNull(testAssertionUntypedAttributeValues);
+    }
+    @Test(dependsOnMethods = {"assertionWithTypedAttributeValuesIsNotNull"})
+    public void candidateUsernamesTypedNotNullOrEmpty() {
+    	testBanner("findSamlAssertionCandidateUsernames finds candidate usernames when they are typed as string");
+    	Set<String> candidateUsernames = SecurityUtils.findSamlAssertionCandidateUsernames(testAssertionTypedAttributeValues, null);
+		Assert.assertNotNull(candidateUsernames);
+		if(null != candidateUsernames)
```

Comment:
Small nitpick but it seems like with the assertion above we don't need the if statement.

Reply:
Yea like I said it was just a nitpick and was something that I noticed. I hadn't thought of someone updating the tests in such a way that the previous assertion would be removed but I guess crazier things have happened :).